src/core_modules/capture-core/rules/actionsCreator/inputHelpers.js

```diff
@@ -31,11 +31,10 @@ function convertFormValuesToClientValues(formValues: {[key: string]: any}, rende
 
 function getValidFormValues(formValues: { [key: string]: any }, fieldsValidation: { [key: string]: boolean }) {
     return Object.keys(formValues)
-        .reduce((accValidFormValues, key) => {
-            const isValid = fieldsValidation[key];
-            accValidFormValues[key] = isValid ? formValues[key] : null;
-            return accValidFormValues;
-        }, {});
+        .filter(key => fieldsValidation[key])
```

Comment:
What used to happen before, is that we iterated over all the form values and returned `null` when there isn't a value. I return now elements that have values. 

Please lets try to think if this will cause any problems in the ling term.

Reply:
LGTM - I would test wether the default value assignment works as before. Check that a "Yes/No" data element left blank is evaluated to "No", or that a Numeric data element defaults to 0.